kuma/users/views.py

```diff
@@ -138,21 +142,115 @@ def ban_user_and_cleanup_summary(request, user_id):
     else:
         user_ban.update(by=request.user, reason='Spam')
 
-    # TODO: In the future this will take the revisions out of request.POST
-    # and either revert them or not. For now list all of the revisions for the past 3 days
     date_three_days_ago = datetime.now().date() - timedelta(days=3)
-    revisions_reverted = user.created_revisions.prefetch_related('document')
-    revisions_reverted = revisions_reverted.defer('content', 'summary').order_by('-id')
-    revisions_reverted = revisions_reverted.filter(created__gte=date_three_days_ago)
+    revisions_from_last_three_days = user.created_revisions.prefetch_related('document')
+    revisions_from_last_three_days = revisions_from_last_three_days.defer('content', 'summary').order_by('-id')
+    revisions_from_last_three_days = revisions_from_last_three_days.filter(created__gte=date_three_days_ago)
+
+    distinct_doc_rev_ids = list(
+        Document.objects.filter(
+            revisions__in=revisions_from_last_three_days
+        ).annotate(
+            latest_rev=Max('revisions')
+        ).values_list('latest_rev', flat=True)
+    )
+    revisions_from_last_three_days_by_distinct_doc = revisions_from_last_three_days.filter(id__in=distinct_doc_rev_ids)
+    # TODO: Phase III: In the future we will take the revisions out of request.POST
+    # and either revert them or not.
+
+    """ The "Actions Taken" section """
+    # The revisions to be submitted to Akismet
+    revisions_to_be_submitted_to_akismet = revisions_from_last_three_days.filter(
+        id__in=request.POST.getlist('revision-id'))
+    # Submit revisions to Akismet as spam
+    submitted_to_akismet = []
+    not_submitted_to_akismet = []
+    for revision in revisions_to_be_submitted_to_akismet:
+        submission = RevisionAkismetSubmission(sender=request.user, type="spam")
+        akismet_submission_data = {'revision': revision.id}
+
+        data = RevisionAkismetSubmissionSpamForm(
+            data=akismet_submission_data,
+            instance=submission,
+            request=request)
+        # Submit to Akismet or note that validation & sending to Akismet failed
+        if data.is_valid():
+            data.save()
+            # Since we only want to display 1 revision per document, only add to
+            # this list if this is one of the revisions for a distinct document
+            submitted_to_akismet.append(revision)
+        else:
+            not_submitted_to_akismet.append(revision)
+    # Find just the latest revision for each document
+    if len(submitted_to_akismet) > 0:
+        distinct_submitted_to_akismet_rev_ids = list(
+            Document.objects.filter(
+                revisions__in=submitted_to_akismet
+            ).annotate(
+                latest_rev=Max('revisions')
+            ).values_list('latest_rev', flat=True)
+        )
```

Comment:
I like this style more than the 110-column monstrosity of ``revisions_from_last_three_days``.  I've also seen this style, borrowed from the JS crowd, which seems to work well for chained queryset operations:

```
    distinct_submitted_to_akismet_rev_ids = list(
        Document.objects
        .filter(revisions__in=submitted_to_akismet)
        .annotate(latest_rev=Max('revisions'))
        .values_list('latest_rev', flat=True))
```